### PR TITLE
Use SYMFONY_ASSETS_INSTALL env variable when calling assets:install

### PR DESCRIPTION
**Pull request by BlackbitDevs**
In https://github.com/pimcore/pimcore/blob/abdca558837f67183845260bc1250ee37936ac52/lib/Composer.php#L251-L258 `assets:install` gets executed without parameters by default (= copy asset files). This can be changed via composer.json's `extra.symfony-assets-install` or env variable `SYMFONY_ASSETS_INSTALL` in https://github.com/pimcore/pimcore/blob/abdca558837f67183845260bc1250ee37936ac52/lib/Composer.php#L190-L193

Imho this is implemented correctly.

In contrast, in https://github.com/pimcore/pimcore/blob/abdca558837f67183845260bc1250ee37936ac52/lib/Tool/AssetsInstaller.php#L64-L83 by default `assets:install` gets executed with `--symlink --relative` because of https://github.com/pimcore/pimcore/blob/abdca558837f67183845260bc1250ee37936ac52/lib/Tool/AssetsInstaller.php#L113-L143

Moreover:
https://github.com/pimcore/pimcore/blob/abdca558837f67183845260bc1250ee37936ac52/lib/Tool/AssetsInstaller.php#L125-L135 does not make sense anyway because 
`$defaults = array_merge([ 'symlink' => true, 'relative' => false ], $defaults)` will overwrite colliding array keys with the second item (see https://www.php.net/manual/de/function.array-merge.php#refsect1-function.array-merge-examples). Thus `$defaults` stays the same anyway.

This PR adjusts `AssetsInstaller` in the same manor as the `Composer` class does it:
1. Environment variable `SYMFONY_ASSETS_INSTALL` has highest priority
2. If 1. is not set, use `extra.symfony-assets-install` from composer.json
3. If 2. is not set, *copy* files instead of *linking*

### CODE REVIEW

File: lib/Tool/AssetsInstaller.php [hunk 1]
```diff
@@ -129,18 +129,32 @@ private function configureOptions(OptionsResolver $resolver)
...
+                    $defaults,
+                    [
+                        'symlink' => true,
+                        'relative' => true,
+                    ]
+                );
             }
         }
 
+        if(in_array(getenv('SYMFONY_ASSETS_INSTALL'), ['symlink', 'relative'])) {
```
Comment (brusch) [inline lib/Tool/AssetsInstaller.php]: Probably better to use `$_SERVER` here, right? Otherwise `.env` won't work?

**Review by brusch — CHANGES_REQUESTED** [review]

File: lib/Tool/AssetsInstaller.php (hunk 1 above)
Comment (BlackbitDevs) [reply lib/Tool/AssetsInstaller.php]: You are right. So also change https://github.com/pimcore/pimcore/blob/abdca558837f67183845260bc1250ee37936ac52/lib/Composer.php#L192 ?

Comment (brusch) [reply lib/Tool/AssetsInstaller.php]: 👍 yep

Comment (BlackbitDevs) [reply lib/Tool/AssetsInstaller.php]: Done

**Comment by BlackbitDevs** [comment]
@brusch Do you know what PhpStan complains about?
Error is
> PHP Warning:  file_get_contents(/home/runner/work/pimcore/pimcore/var/cache/test/App_KernelTestDebugContainer.xml): Failed to open stream: No such file or directory in /home/runner/work/pimcore/pimcore/vendor/phpstan/phpstan-symfony/src/Symfony/XmlServiceMapFactory.php on line 29
Warning: file_get_contents(/home/runner/work/pimcore/pimcore/var/cache/test/App_KernelTestDebugContainer.xml): Failed to open stream: No such file or directory in /home/runner/work/pimcore/pimcore/vendor/phpstan/phpstan-symfony/src/Symfony/XmlServiceMapFactory.php on line 29
> 
> In XmlServiceMapFactory.php line 31:                                                                             
  Container /home/runner/work/pimcore/pimcore/var/cache/test/App_KernelTestDe  
  bugContainer.xml does not exist      

I can hardly believe that this PR's changes are responsible for that. Have tried merging current `10.5` branch but to no success.

**Comment by brusch** [comment]
@BlackbitDevs I guess the error is: 
```
Script Pimcore\Composer::clearCache handling the pimcore-scripts event terminated with an exception
Error: Undefined array key "SYMFONY_ASSETS_INSTALL"
```

File: lib/Composer.php [hunk 2]
```diff
@@ -216,7 +216,7 @@ protected static function getPhpArguments()
             $arguments = $phpFinder->findArguments();
         }
 
-        if ($env = getenv('COMPOSER_ORIGINAL_INIS')) {
+        if ($env = $_SERVER['COMPOSER_ORIGINAL_INIS']) {
```
Comment (brusch) [inline lib/Composer.php]:
```suggestion
        if ($env = $_SERVER['COMPOSER_ORIGINAL_INIS'] ?? null) {
```

File: lib/Composer.php [hunk 3]
```diff
@@ -234,8 +234,8 @@ protected static function getOptions(Event $event)
     {
         $options = array_merge(static::$options, $event->getComposer()->getPackage()->getExtra());
 
-        $options['symfony-assets-install'] = getenv('SYMFONY_ASSETS_INSTALL') ?: $options['symfony-assets-install'];
-        $options['symfony-cache-warmup'] = getenv('SYMFONY_CACHE_WARMUP') ?: $options['symfony-cache-warmup'];
+        $options['symfony-assets-install'] = $_SERVER['SYMFONY_ASSETS_INSTALL'] ?: $options['symfony-assets-install'];
+        $options['symfony-cache-warmup'] = $_SERVER['SYMFONY_CACHE_WARMUP'] ?: $options['symfony-cache-warmup'];
```
Comment (brusch) [inline lib/Composer.php]:
```suggestion
        $options['symfony-assets-install'] = $_SERVER['SYMFONY_ASSETS_INSTALL'] ?? $options['symfony-assets-install'];
        $options['symfony-cache-warmup'] = $_SERVER['SYMFONY_CACHE_WARMUP'] ?? $options['symfony-cache-warmup'];
```

File: lib/Tool/AssetsInstaller.php [hunk 4]
```diff
@@ -129,18 +129,32 @@ private function configureOptions(OptionsResolver $resolver)
...
+                    $defaults,
+                    [
+                        'symlink' => true,
+                        'relative' => true,
+                    ]
+                );
             }
         }
 
+        if(in_array($_SERVER['SYMFONY_ASSETS_INSTALL'], ['symlink', 'relative'])) {
```
Comment (brusch) [inline lib/Tool/AssetsInstaller.php]:
```suggestion
        if(in_array($_SERVER['SYMFONY_ASSETS_INSTALL'] ?? null, ['symlink', 'relative'])) {
```

File: lib/Tool/AssetsInstaller.php [hunk 5]
```diff
@@ -129,18 +129,32 @@ private function configureOptions(OptionsResolver $resolver)
...
+            $defaults['symlink'] = true;
+        }
+
+        if($_SERVER['SYMFONY_ASSETS_INSTALL'] === 'relative') {
```
Comment (brusch) [inline lib/Tool/AssetsInstaller.php]:
```suggestion
        if($_SERVER['SYMFONY_ASSETS_INSTALL'] ?? null === 'relative') {
```

**Review by brusch — CHANGES_REQUESTED** [review]

File: lib/Composer.php (hunk 3 above)
Comment (BlackbitDevs) [reply lib/Composer.php]: Have fixed it with `!empty($_SERVER['SYMFONY_ASSETS_INSTALL'])` - is also easier to read.

File: lib/Composer.php (hunk 2 above)
Comment (BlackbitDevs) [reply lib/Composer.php]: Have fixed it with `!empty($_SERVER['SYMFONY_ASSETS_INSTALL'])` - is also easier to read.

File: lib/Tool/AssetsInstaller.php (hunk 4 above)
Comment (BlackbitDevs) [reply lib/Tool/AssetsInstaller.php]: Done

File: lib/Tool/AssetsInstaller.php (hunk 5 above)
Comment (BlackbitDevs) [reply lib/Tool/AssetsInstaller.php]: Done.

**Review by mattamon — APPROVED** [review]
LGTM

**Comment by brusch** [comment]
Thanks a lot!

**Comment by brusch** [comment]
This is changing the default behavior, so I'd say we need to revert this in `10.5` and apply it to `11.x`.

**Comment by BlackbitDevs** [comment]
@brusch Ok, for me it was a bugfix. But does not matter to me. Do you care for reverting it in 10.5 branch and putting it to 11?

**Comment by brusch** [comment]
@BlackbitDevs I've created a follow up ticket for us, but PRs are of course more than welcome 😊 Thanks!

**Comment by BlackbitDevs** [comment]
Have currently quite a lot to do. But will keep this in mind if I see some free space...

**Comment by aryaantony92** [comment]
> Have currently quite a lot to do. But will keep this in mind if I see some free space...

@BlackbitDevs May I know if you have some time  for creating the 11.x PR or I can work on it :). Thanks!

**Comment by BlackbitDevs** [comment]
@aryaantony92 Would be great if you could care about this.

**Comment by aryaantony92** [comment]
> @aryaantony92 Would be great if you could care about this.

Sure, Thanks

**Pull request merged**